src/test/java/POM/pages/BasePage.java

```diff
@@ -5,7 +5,9 @@
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.interactions.Actions;
 import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
 import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
 
 import java.time.Duration;
 
```

Comment:
**I recommend:** make the variables protected, because protected access gives the subclass a chance to use the helper method or variable, while preventing a nonrelated class from trying to use it.

```
  protected WebDriver driver;
  protected WebDriverWait wait;
  protected Actions actions;
```